controller/volume_controller.go

```diff
@@ -1657,6 +1662,10 @@ func (vc *VolumeController) checkForAutoDetachment(v *longhorn.Volume, e *longho
 
 	// Do auto-detachment for non-restore/DR volumes.
 	if !v.Status.RestoreRequired && !v.Status.IsStandby {
+		// Don't for auto-detachment if the evivtion is going on.
```

Comment:
Put it outside of this `if`. e.g.:
```
	if v.Status.ExpansionRequired {
		return nil
	}
       
        if vc.hasReplicaEvictionRequested(rs) {
		return nil
	}		
```

And there is a typo `evivtion` in the comment.

Reply:
Fixed, thanks.